File: src/main/kotlin/org/kite9/diagram/logging/LogicException.java
package org.kite9.diagram.logging;

/**
 * LogicExceptions are thrown to represent an underlying bug in Kite9
 * Visualization (as opposed to user-error).
 * 
 * @author robmoffat
 *
 */
public class LogicException extends RuntimeException {

	public LogicException() {
		super();
	}

	public LogicException(String arg0, Throwable arg1) {
		super(arg0, arg1);
	}

	public LogicException(String arg0) {
		super(arg0);
	}

	public LogicException(Throwable arg0) {
		super(arg0);
	}

}

File: src/main/kotlin/org/kite9/diagram/logging/LogicException.kt
package org.kite9.diagram.logging

import java.lang.RuntimeException

/**
 * LogicExceptions are thrown to represent an underlying bug in Kite9
 * Visualization (as opposed to user-error).
 *
 * @author robmoffat
 */
open class LogicException : RuntimeException {
    constructor() : super() {}
    constructor(arg0: String?, arg1: Throwable?) : super(arg0, arg1) {}
    constructor(arg0: String?) : super(arg0) {}
    constructor(arg0: Throwable?) : super(arg0) {}
}